activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb

```diff
@@ -93,6 +93,17 @@ def column_exists?(table_name, column_name, type = nil, options = {})
                                       (!options.key?(:null)      || c.null == options[:null]) }
       end
 
+      # Returns just a table's primary key
+      def primary_key(table_name)
+        pks = primary_keys(table_name)
+        return nil if pks.nil? || pks.size != 1
```

Comment:
If all `primary_keys` return array, we can write `return nil if pks.size != 1` here.